autoprecompiles/src/lib.rs

```diff
@@ -184,9 +184,12 @@ impl<T: Display> Display for SymbolicMachine<T> {
     }
 }
 
-impl<T: Display> SymbolicMachine<T> {
+impl<T: Display + Ord + Clone> SymbolicMachine<T> {
     pub fn render(&self, bus_map: &BusMap) -> String {
-        let mut output = String::new();
+        let mut output = format!(
+            "// Symbolic machine using {} unique columns\n",
```

Comment:
```suggestion
            "// Symbolic machine using {} unique main columns\n",
```
There might also be second-stage columns related to the bus argument.